stc_game/Assets/Scripts/ReviveUI/ReviveController.cs

```diff
@@ -9,20 +9,25 @@ public class ReviveController : MonoBehaviour
     public GameObject reviveAtEntrance;
     public Canvas reviveCanvas;
     public bool reviveInTown;
+    private SceneController sceneController;
 
     private void Awake()
     {
         reviveCanvas = gameObject.GetComponent<Canvas>();
         reviveCanvas.enabled = false;
         reviveInTown = false;
+        sceneController = SceneController.Instance;
     }
 
     private void Update()
     {
-        if (SceneController.Instance.currentSceneName.Equals(GameStrings.Scenes.TownScene))
-            reviveAtEntrance.SetActive(false);
-        else
-            reviveAtEntrance.SetActive(true);
+        if (sceneController)

```

Comment:
why is this in the Update function?

Reply:
oops, changed it to subscribe to the AfterSceneLoad event instead